state-machine.md

```diff
@@ -55,30 +58,50 @@ func ApplyMessage(st StateTree, msg Message) MessageReceipt {
 		Fatal("invalid nonce")
 	}
 
+  toActor, found := st.GetActor(msg.To)
+  if !found {
+    toActor = TryCreateAccountActor(st, msg.To)
+  }
+  
 	st.DeductFunds(msg.From, totalCost)
 	st.DepositFunds(msg.To, msg.Value)
 
 	vmctx := makeVMContext(st, msg)
 
-	ret, errcode := fromActor.Invoke(vmctx, msg.Method, msg.Params)
-	if errcode != 0 {
-		// revert all state changes since snapshot
-		st.Revert()
-		st.DeductFunds(msg.From, vmctx.GasUsed() * msg.GasPrice)
-	} else {
-		// refund unused gas
-		st.DepositFunds(msg.From, (msg.GasLimit - vmctx.GasUsed()) * msg.GasPrice)
-	}
+  if msg.Method != "" {
+		ret, errcode := toActor.Invoke(vmctx, msg.Method, msg.Params)
+		if errcode != 0 {
+			// revert all state changes since snapshot
+			st.Revert()
+			st.DeductFunds(msg.From, vmctx.GasUsed() * msg.GasPrice)
+		} else {
+			// refund unused gas
+			st.DepositFunds(msg.From, (msg.GasLimit - vmctx.GasUsed()) * msg.GasPrice)
+		}
+  }
 
 	// reward miner gas fees
-	st.DepositFunds(msg.To, msg.GasPrice * vmctx.GasUsed())
+	st.DepositFunds(BlockMiner, msg.GasPrice * vmctx.GasUsed())
```

Comment:
other big fix was giving the gas reward to the miner, not the actor youre sending the message to